src/gui/editorwidgets/qgsrelationreferencewidget.cpp

```diff
@@ -155,7 +156,9 @@ QgsRelationReferenceWidget::~QgsRelationReferenceWidget()
 {
   deleteHighlight();
   unsetMapTool();
-  delete mMapTool;
+  unsetMapTool();
+  delete mMapToolIdentify;
+  delete mMapToolDigitize;
```

Comment:
Better use a `QObjectUniquePtr`?

Reply:
Done.
Just curious, why using a QObjectUniquePtr instead of a std::unique_ptr or a QScopedPointer ?

Reply:
QObjects can have a parent and if they have, they are owned by it. If the parent object is deleted first, other smart pointers are left pointing to an already deleted object.

Reply:
OK, thks